sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -512,6 +512,21 @@ assert typeConversionAllowed(cudfCv, type) : "Type conversion is not allowed fro
     return new GpuColumnVector(type, cudfCv);
   }
 
+  /**
+   * Converts a cudf internal vector to a Spark compatible vector. No reference counts
+   * are incremented so you need to either close the returned value or the input value,
+   * but not both. This conversion performs an unconditional check that the types are
+   * convertible rather than an assertion check.
+   * @throws IllegalArgumentException if the type conversion check fails
+   */
+  public static GpuColumnVector fromChecked(ai.rapids.cudf.ColumnVector cudfCv, DataType type) {
+    if (!typeConversionAllowed(cudfCv, type)) {
```

Comment:
Please update the javadocs on the `typeConversionAllowed` method.

```
  /**
   * This should only ever be called from an assertion.
   */
```

is no longer true.  The message we have right now is probably OK, but it might be worth refactoring `typeConversionAllowed` in the future so it can explain exactly why it is not allowing the conversion, to help with users debugging issues.

Reply:
Done.  I'll file a followup issue to track improving the message for users.